### PR TITLE
(chore): add e2e tests for encryption-at-rest

### LINKED ISSUE #8405: Add e2e test cases for encryption-at-rest
**User Story**
If encryption-at-rest is not working properly, using it puts users at risk of ending up with encrypted data in etcd that cannot be decrypted. Therefore, we need solid e2e test cases that test encryption-at-rest for several scenarios:

- Enabling encryption-at-rest on an existing cluster.
- Rotating a static key.
- Rotating a KMS key reference.
- Disabling encryption-at-rest on a cluster.
- Restoring an etcd backup that was created with a previous key.

**Acceptance Criteria**
A comprehensive e2e test case set is added and tests the scenarios mentioned above to deliver a solid encryption-at-rest feature.

### PR DESCRIPTION
**Pull request by buraksekili**
**What this PR does / why we need it**:

This commit adds e2e tests for the encryption-at-rest functionality in KKP. The changes include:

- Add new e2e test package for encryption-at-rest with test cases for enabling and disabling encryption
- Create a CI script to run the encryption-at-rest e2e tests
- Update Prow configuration to add encryption-at-rest e2e tests to the CI pipeline
- Fix parameter handling in the encryption-at-rest controller
- Add verification for proper encryption and decryption of Kubernetes secrets

The tests validate that:
1. Enabling encryption works correctly and data gets encrypted
2. Disabling encryption works correctly and data gets decrypted
3. API server configuration is properly updated with encryption settings

**Which issue(s) this PR fixes**:

Fixes #8405

**What type of PR is this?**

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
NONE
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: .prow/features.yaml [hunk 1]
```diff
@@ -308,3 +308,39 @@ presubmits:
               cpu: 2
             limits:
               memory: 6Gi
+
+  - name: pre-kubermatic-encryption-at-rest-e2e
+    run_if_changed: "(pkg/controller/seed-controller-manager/encryption-at-rest-controller/|pkg/test/e2e/encryption-at-rest/)"
```
Comment (embik) [inline .prow/features.yaml]: I'm wondering if this should be run more frequently than with those files, e.g. when we make changes to the Kubernetes version or the apiserver.

File: pkg/test/e2e/encryption-at-rest/ear_test.go [hunk 2]
```diff
@@ -0,0 +1,464 @@
...
+
+	err = wait.PollImmediateLog(
+		ctx, r.logger, defaultInterval, defaultTimeout,
+		func(ctx context.Context) (transient error, terminal error) {
+			c := cluster.DeepCopy()
+			if err := r.seedClient.Get(ctx, ctrlruntimeclient.ObjectKeyFromObject(c), c); err != nil {
+				return fmt.Errorf("failed to get cluster: %w", err), nil
+			}
+
+			r.logger.Infof("cluster status: %+v", c.Status)
```
Comment (embik) [inline pkg/test/e2e/encryption-at-rest/ear_test.go]: I think this might be a bit much, maybe it's a leftover from development debugging?

File: pkg/test/e2e/encryption-at-rest/ear_test.go [hunk 3]
```diff
@@ -0,0 +1,464 @@
...
```
Comment (embik) [inline pkg/test/e2e/encryption-at-rest/ear_test.go]: same here.

**Review by embik — COMMENTED** [review]
Some minor nits.

File: .prow/features.yaml (hunk 1 above)
Comment (buraksekili) [reply .prow/features.yaml]: Makes sense! Do you think we should run this test case against every PR?

File: pkg/test/e2e/encryption-at-rest/ear_test.go (hunk 3 above)
Comment (buraksekili) [reply pkg/test/e2e/encryption-at-rest/ear_test.go]: thanks! removed these logs in 19839f8a79cc82b7d06b8a34d23949a35a2deb1f

File: .prow/features.yaml (hunk 1 above)
Comment (embik) [reply .prow/features.yaml]: Every commit might be a bit much, but I think it makes sense to test it against a) any apiserver changes and b) any Kubernetes version changes (which is somewhere in `pkg/defaulting`, afaik?).

Comment (buraksekili) [reply .prow/features.yaml]: Sounds good! I’m planning to run this test case if there are changes in `pkg/defaulting` or `pkg/resources`. Does that approach work for you?

Comment (embik) [reply .prow/features.yaml]: I think `pkg/resources/apiserver` should be sufficient, otherwise lgtm!

Comment (buraksekili) [reply .prow/features.yaml]: Thanks for the review. Updated accordingly in 238d6da05ff1a1f2b0a28bb605a3200635034771

Comment (embik) [reply .prow/features.yaml]: Sorry, I wasn't clear in what I meant to say - your previous triggers were perfect, the addition of `pkg/defaulting` makes sense, I just meant to say instead of `pkg/resources` you can "just" add `pkg/resources/apiserver`. Not replace everything with it, but I can see how it could be read that way.

Comment (buraksekili) [reply .prow/features.yaml]: oh, my bad. i misunderstood, thanks for the clarification :) 
i've updated the PR accordingly.

**Comment by embik** [comment]
The issue (#8405) mentions more cases to test for, and with EAR it's important to test edge cases to understand if things could go wrong.

If you would like to keep things as is, can you please update the PR description that it doesn't automatically close #8405 (and say something like "Towards #8405") so we can implement more test cases later on?

**Comment by buraksekili** [comment]
@embik, apart from the KMS, which is not supported according to docs, I've added new test cases to cover cluster backup scenarios. could you please have a look again?

File: pkg/test/e2e/encryption-at-rest/ear_test.go [hunk 4]
```diff
@@ -0,0 +1,906 @@
...
+		seedClient: seedClient,
+		userClient: userClient,
+		config:     config,
+		logger:     logger,
+		testJig:    testJig,
+	}
+
+	// Test Case 1:
+	// Enable encryption at rest and verify it works
+	logger.Info("Test Case 1 running...")
```
Comment (embik) [inline pkg/test/e2e/encryption-at-rest/ear_test.go]: A minor style note, given this is running end-to-end as what I would call a single test case (we aren't doing table testing or anything like that), I would suggest changing the log lines for this (and all the other "Test Case x running"). The comments above each lines are explaining well what is being tested, maybe we can put that information (in abbreviated form) into the logs?

Something like

```suggestion
	logger.Info("Enabling encryption at rest...")
```

And similar for later log lines.

**Review by embik — CHANGES_REQUESTED** [review]

File: pkg/test/e2e/encryption-at-rest/ear_test.go (hunk 4 above)
Comment (buraksekili) [reply pkg/test/e2e/encryption-at-rest/ear_test.go]: Thanks! removed redundant log statements in 6379af519a1da088688f2a62047439ad73d818c5. let me know if it needs further work

**Review by embik — APPROVED** [review]
/approve

🎉
